tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+
+The following code illustrates how to create, save, and schedule the fibonacci job:
+
+```python
+from girder.utility.model_importer import ModelImporter
+
+...
+
+# get the job model singleton which uses the factory method createJob
+jobModel = ModelImporter.model('job', 'jobs')
+
+# Create the job
+job = jobModel.createJob(
+    title='<NAME OF THE JOB>',
+    type='<JOB TYPE>', handler='worker_handler',
+    user=getCurrentUser(), public=False, args=(20,), kwargs={},
+    otherFields={
+        'celeryTaskName': 'my_custom_package.tasks.fibonacci'
+    })
+
+# Save the job
+jobModel.save(job)
+
+# Schedule the job
+jobModel.scheduleJob(job)
+
+return job
+```
+
+*This is code you might see in a REST request handler that is dedicated to launching fibonacci jobs. See the [Jobs](http://girder.readthedocs.io/en/latest/plugins.html?highlight=schedule#jobs) documentation for more information about creating, saving and scheduling jobs.*
+
+In the above code we get the job model singleton which uses a [factory](https://en.wikipedia.org/wiki/Factory_method_pattern) method ```createJob(...)``` to create the job. In our example you should replace the ```title``` and ```type``` keyword arguments with values that make sense for your purposes. Note that ```type``` is a arbitrary string used to categorize jobs, it has no inherent semantic value. ```user``` should be an instance of a user model,  the job will 'belong' to this user. ```public``` effects whether or not only the ```user``` can see the job or not. ```args``` Is a list of arguments to apply the consumer-side task object too (In our example ```n``` in ```my_custom_package.tasks.fibonacci``` will be 20). ```kwargs``` is a list of keyword arguments to apply the consumer-side task object too (In our example none). ```handler``` in our case *must be* ```"worker_handler"``` or the Worker plugin handler will not execute and the Celery message will not be produced. Finally ```otherFields``` is a dictionary of fields that are specific to Worker plugin job models. 
+
+Currently the Worker plugin supports two ```otherFields```:
+
++ **celeryTaskName** should be the name of the task object. This is almost always the python importable path to the task as a string (unless you have specifically set the [name](http://docs.celeryproject.org/en/latest/userguide/tasks.html#names) when decorating the function). In our example this is the string "my_custom_package.tasks.fibonacci". If you do not specify a ```celeryTaskName``` in ```otherFields``` it will use "girder_worker.run" by default.
++ **celeryQueue** allows you to place the produces message on a different RabbitMQ queue than the default ```celery``` queue.
```

Comment:
the only reason you would want to do this is because you are using an advanced feature of Celery/RabbitMQ.  Most people won't need this,  but i didn't want to leave it out entirely